File: Algorithms and Data  Structure Problems/Arrays/1) firstDuplicate/firstDuplicate.cs
int firstDuplicate(int[] a) {
    int value = -1;
    Dictionary<int, int> nums = new Dictionary<int,int>();
    for(int i = 0; i < a.Length; i++){
        if(nums.ContainsKey(a[i])){
            value = a[i];
            break;
        }
        nums.Add(a[i], 1);
    }
    return value;
}

File: Algorithms and Data  Structure Problems/Arrays/1) firstDuplicate/firstDuplicate.cpp
int firstDuplicate(std::vector<int> a) {
    int value = -1;
    std::unordered_map<int,int>nums;
    for(int i = 0; i < a.size(); i++){
        if(nums.find(a[i]) != nums.end()){
                value = a[i];
                break;
            }
        nums[a[i]] = 1;
        }        
    return value;
}